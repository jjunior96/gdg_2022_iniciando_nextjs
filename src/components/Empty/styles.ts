import styled, { css } from 'styled-components';

export const Container = styled.div`
  width: 100%;

  display: flex;
  align-items: center;
`;

export const Content = styled.div`
  ${({ theme }) => css`
    margin: 0 auto;

    display: flex;
    align-items: center;
    flex-direction: column;

    width: 100%;
    max-width: 40rem;

    svg {
      font-size: ${theme.font.sizes.huge};
      color: ${theme.colors.gray_200};
    }
  `}
`;

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    color: ${theme.colors.gray_300};

    margin: ${theme.spacings.xsmall} 0;
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    color: ${theme.colors.gray_300};

    max-width: 100%;
    word-wrap: break-word;

    margin-bottom: ${theme.spacings.medium};
  `}
`;
