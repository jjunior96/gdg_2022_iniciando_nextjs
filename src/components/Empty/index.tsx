import { FiInbox as EmptyIcon } from 'react-icons/fi';

import Button from 'components/Button';

import * as S from './styles';

export type EmptyProps = {
  title?: string;
  description?: string;
  hasLink?: boolean;
  icon?: boolean;
};

const Empty = ({
  title,
  description,
  icon = true,
  hasLink = false
}: EmptyProps) => {
  return (
    <S.Container>
      <S.Content>
        {icon && <EmptyIcon data-testid="empty-icon" />}

        <S.Title>{title ? title : 'No data'}</S.Title>
        {description && <S.Description>{description}</S.Description>}

        {hasLink && (
          <Button href="/home" as="a">
            Go to home
          </Button>
        )}
      </S.Content>
    </S.Container>
  );
};

export default Empty;
