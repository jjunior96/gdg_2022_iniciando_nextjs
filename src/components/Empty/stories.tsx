import { Story, Meta } from '@storybook/react/types-6-0';

import Empty, { EmptyProps } from '.';

export default {
  title: 'Empty',
  component: Empty
} as Meta;

export const Default: Story<EmptyProps> = (args) => <Empty {...args} />;

Default.args = {
  title: 'No data',
  description: 'No data for to show',
  hasLink: true
};
