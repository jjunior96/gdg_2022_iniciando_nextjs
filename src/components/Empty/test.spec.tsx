import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/__config__/helpers';

import Empty from '.';

describe('<Empty />', () => {
  it('should render default', () => {
    const { container } = renderWithTheme(<Empty />);

    const titleComponent = screen.getByText(/no data/i);

    expect(titleComponent).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly', () => {
    renderWithTheme(<Empty title="title" />);

    const titleComponent = screen.getByText(/title/);
    const iconComponent = screen.getByTestId(/empty-icon/);

    expect(titleComponent).toBeInTheDocument();
    expect(iconComponent).toBeInTheDocument();
  });

  it('should render with description', () => {
    renderWithTheme(<Empty title="title" description="description" />);

    const titleComponent = screen.getByText(/title/i);
    const descriptionComponent = screen.getByText(/description/i);

    expect(titleComponent).toBeInTheDocument();
    expect(descriptionComponent).toBeInTheDocument();
  });

  it('should render with link for home', () => {
    renderWithTheme(<Empty title="title" hasLink />);

    expect(screen.getByText(/title/)).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /go to home/i })
    ).toBeInTheDocument();
  });

  it('should render without icon when false passed', () => {
    renderWithTheme(<Empty title="title" icon={false} />);

    const iconComponent = screen.queryByTestId(/empty-icon/);

    expect(iconComponent).not.toBeInTheDocument();
  });
});
