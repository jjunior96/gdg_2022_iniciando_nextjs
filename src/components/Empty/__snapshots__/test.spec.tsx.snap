// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> should render default 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  max-width: 40rem;
}

.c1 svg {
  font-size: 6.4rem;
  color: #91AAB4;
}

.c2 {
  font-size: 3.2rem;
  color: #3E606F;
  margin: 1.6rem 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      data-testid="empty-icon"
      fill="none"
      height="1em"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="22 12 16 12 14 15 10 15 8 12 2 12"
      />
      <path
        d="M5.45 5.11L2 12v6a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2v-6l-3.45-6.89A2 2 0 0 0 16.76 4H7.24a2 2 0 0 0-1.79 1.11z"
      />
    </svg>
    <h2
      class="c2"
    >
      No data
    </h2>
  </div>
</div>
`;
